sdk/storage/Azure.Storage.Blobs/tests/UnitTests.cs

```diff
@@ -18,5 +20,34 @@ public void BlobDownloadInfo_Dispose()
             blobDownloadInfo.Dispose();
             Assert.IsTrue(stream.IsDisposed);
         }
+
+        [Test]
+        public void GetBlobOptions_VariousFlagCombos()
+        {
+            GetBlobOptions options = GetBlobOptions.Metadata | GetBlobOptions.DeletedBlobs;
+            Assert.AreEqual(
+                "deleted,metadata",
+                string.Join(",", System.Linq.Enumerable.Select(options.AsIncludeItems(), item => Azure.Storage.Blobs.BlobRestClient.Serialization.ToString(item))));
+
+            options = GetBlobOptions.Metadata | GetBlobOptions.DeletedBlobs | GetBlobOptions.None;
+            Assert.AreEqual(
+                "deleted,metadata",
+                string.Join(",", System.Linq.Enumerable.Select(options.AsIncludeItems(), item => Azure.Storage.Blobs.BlobRestClient.Serialization.ToString(item))));
+
+            options = GetBlobOptions.CopyOperationStatus | GetBlobOptions.Metadata | GetBlobOptions.Snapshots;
+            Assert.AreEqual(
+                "copy,metadata,snapshots",
+                string.Join(",", System.Linq.Enumerable.Select(options.AsIncludeItems(), item => Azure.Storage.Blobs.BlobRestClient.Serialization.ToString(item))));
+
+            options = GetBlobOptions.Metadata | GetBlobOptions.CopyOperationStatus | GetBlobOptions.Snapshots | GetBlobOptions.UncommittedBlobs | GetBlobOptions.DeletedBlobs;
+            Assert.AreEqual(
+                "copy,deleted,metadata,snapshots,uncommittedblobs",
```

Comment:
Nit - This ordering is kind of weird.  I think metadata should be first in your `ToIncludeItems` method.  Then it's "blob details" followed by "types of blobs".

Reply:
Alphabetical order is required by the service for signing to work.. at least according to the existing comment in the AsIncludeItems method.

Reply:
Oh - good catch.